src/plots/cartesian/axes.js

```diff
@@ -2636,6 +2649,31 @@ axes.drawOne = function(gd, ax, opts) {
     return Lib.syncOrAsync(seq);
 };
 
+function filterPush(push, automargin) {
+    if(!push) return push;
+
+    var keepMargin = [];
+
+    Object.keys(MARGIN_MAPPING).forEach(function(key) {
+        if(automargin.indexOf(key) !== -1) {
+            MARGIN_MAPPING[key].forEach(function(item) {
```

Comment:
Can't we simply use one [`concact`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/concat) here instead of the inner loop and push?